lib/rrd.ml

```diff
@@ -163,6 +156,16 @@ let copy_rrd x =
     rrd_rras = Array.map copy_rra x.rrd_rras;
   }
 
+(* CA-329043: avoid producing out-of-range rates *)
+let cf_init_value cf ds =
+  let default = match cf with
+  | CF_Average -> 0.0
+  | CF_Min -> infinity
+  | CF_Max -> neg_infinity
+  | CF_Last -> nan
+  in
+  min ds.ds_max (max ds.ds_min default)
```

Comment:
This will still let the `nan` propagate through, but I assume that is fine, since `nan` here means unknown?

Reply:
Indeed